susan/db/rdbms/dhcp.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2017 <thenakliman@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from susan.db import dhcp as dhcp_db
+from susan.db.rdbms.models import dhcp as d_model
+from susan.db import rdbms
+
+
+class Subnet(dhcp_db.DHCPdb):
+    def __init__(self):
+        super(Subnet, self).__init__()
+
+    def create_subnet(self, network, cidr, gateway=None):
+        session = rdbms.get_session()
+        row = d_model.Subnet(network=network, cidr=cidr, gateway=gateway)
+        session.add(row)
+        session.flush()
+
+    def update_subnet(self, id_, network=None, cidr=None, gateway=None):
+        pass
+
+    def delete_subnet(self, id_):
+        session = rdbms.get_session()
+        session.query(d_model.Subnet).filter_by(id=id_).delete()
+
+    @staticmethod
+    def get_subnet(id_):
+        session = rdbms.get_session()
+        return session.query(d_model.Subnet).filter_by(id=id_).one_or_none()
+
+
+class Range(dhcp_db.DHCPdb):
+    def __init__(self):
+        super(Range, self).__init__()
+
+    def create_range(self, subnet_id, start_ip, end_ip):
+        session = rdbms.get_session()
+        row = d_model.IPRange(subnet_id=subnet_id, start_ip=start_ip,
+                              end_ip=end_ip)
+        session.add(row)
+        session.flush()
+
+    def update_range(self, subnet_id, start_ip=None, end_ip=None):
+        pass
+
+    @staticmethod
+    def delete_range(subnet_id):
+        d_model.IPRange(d_model.IPRange).filter_by(subnet_id=subnet_id).delete()
+
+    @staticmethod
+    def get_range(subnet_id, start_ip, end_ip):
```

Comment:
I am really not sure of the purpose of this method, how someone will get start and end data and if they have it, why do they need to connect to database.